### PR TITLE
Fix to issue 625 libdl.so.2 DSO missing

### PR DESCRIPTION
**Pull request by waltercool**
Fix proposal to https://github.com/ethereum-mining/ethminer/issues/625

### CODE REVIEW

**Comment by smurfy** [comment]
Link does not work on windows. So you need to put it into a "only unix" check.

**Comment by waltercool** [comment]
Thanks, now I added dl dependency only for UNIX systems.

File: libhwmon/CMakeLists.txt [hunk 1]
```diff
@@ -6,10 +6,14 @@ set(SOURCES
 )
 
 add_library(hwmon ${SOURCES})
-target_link_libraries(hwmon devcore)
+if (UNIX)
+	target_link_libraries(hwmon devcore dl)
```
Comment (reagentoo) [inline libhwmon/CMakeLists.txt]: I suggest using CMake's list APPEND before target_link_libraries.

Comment (waltercool) [reply libhwmon/CMakeLists.txt]: @reagentoo Sorry, not really CMake user, would this new CR proposal works? At least it compiled to me

**Comment by waltercool** [comment]
@reagentoo Would this new CR work? Not really CMake user.

File: libhwmon/CMakeLists.txt [hunk 2]
```diff
@@ -6,10 +6,15 @@ set(SOURCES
 )
 
 add_library(hwmon ${SOURCES})
+
```
Comment (reagentoo) [inline libhwmon/CMakeLists.txt]: `set(HWMON_LINK_LIBRARIES devcore)`

File: libhwmon/CMakeLists.txt [hunk 3]
```diff
@@ -6,10 +6,15 @@ set(SOURCES
...
+if (UNIX)
+	target_link_libraries(hwmon dl)
```
Comment (reagentoo) [inline libhwmon/CMakeLists.txt]: `list(APPEND HWMON_LINK_LIBRARIES dl)`

File: libhwmon/CMakeLists.txt [hunk 4]
```diff
@@ -6,10 +6,15 @@ set(SOURCES
...
+endif ()
+
 target_link_libraries(hwmon devcore)
```
Comment (reagentoo) [inline libhwmon/CMakeLists.txt]: `target_link_libraries(hwmon ${HWMON_LINK_LIBRARIES})`

**Comment by waltercool** [comment]
@reagentoo Changes added. Thanks

**Comment by PikkuJose** [comment]
Is there any chance of this backported to 0.13?

Thanks!

**Comment by smurfy** [comment]
@PikkuJose

isn't that just an compile issue? If so you can just copy the the few lines of this pr into your source

**Comment by PikkuJose** [comment]
Well, I sometimes like to be lazy:

https://github.com/PikkuJose/foxiverlay/blob/master/net-p2p/ethminer/ethminer-0.13.0.ebuild

Nothing easier than just emerge it :)

**Pull request merged**